src/PolynomialFunction.java

```diff
@@ -123,33 +126,31 @@ public void derive() {
         functionString =  sb.toString();
     }
 
-    public List<Double> getZeroPoints(double start, double end, double step) {
-        List<Double> zeros = new ArrayList<>();
+    private void calcRoots(double start, double end, double step) {
```

Comment:
this should return the list of roots, directly.

Reply:
Done